lib/cog_api/fake/server.ex

```diff
@@ -20,41 +20,65 @@ defmodule CogApi.Fake.Server do
     end)
   end
 
-  def index(resource_name) do
+  def index(%{key: resource_name, associations: associations}) do
+    find_all(resource_name)
+    |> Enum.map(fn resource -> expand_assocations(resource, associations) end)
+  end
+
+  defp find_all(resource_name) do
     Agent.get(__MODULE__, fn server ->
       Map.fetch!(server, resource_name)
     end)
   end
 
-  def show(resource_name, id) do
-    Agent.get(__MODULE__, fn server ->
+  def show(%{key: resource_name, associations: associations}, id) do
+    resource = Agent.get(__MODULE__, fn server ->
       find_by_id(server, resource_name, id)
     end)
+
+    if resource do
+      expand_assocations(resource, associations)
+    end
   end
 
-  def show_by_key(resource_name, key, value) do
+  def raw_show(%{key: resource_name}, id) do
```

Comment:
This is only for testing purposes but I couldn't find any way to get this information in the test otherwise.